sw/device/tests/spi_host_smoketest.c

```diff
@@ -0,0 +1,96 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_spi_host.h"
+#include "sw/device/lib/runtime/hart.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/runtime/print.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_framework/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+
+const test_config_t kTestConfig;
+
+void hexdump(void *addr, size_t len) {
```

Comment:
Can we put this in a central location? I would accept a `printf` patch to add something like `%y` and `%Y` for this purpose, with similar behavior to the current `%z`.

Reply:
I can send a follow-up PR to move this to something like `sw/device/lib/runtime` and possibly integrate it into `printf`.

Reply:
Sounds good.